### PR TITLE
Create release workflow

### PR DESCRIPTION
**Pull request by tannalynn**
This workflow automates the release for this gem on push to main. https://github.com/newrelic/newrelic-telemetry-sdk-ruby/issues/16
This will build the gem, determine the version, tag the version on github, and push to rubygems.
I copied the scripts over from the agent for releasing to use in this workflow as well.
The secrets used have already been added to the repo.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - name: Configure gem credentials
+      run: |
+        echo "GEM_HOST_API_KEY=${{ secrets.RUBYGEMS_API_KEY }}" >> $GITHUB_ENV
+        echo "RUBYGEMS_MFA_KEY=${{ secrets.RUBYGEMS_MFA_KEY }}" >> $GITHUB_ENV
+
+    - name: Build newrelic-telemetry_sdk gem
+      run: gem build newrelic-telemetry_sdk.gemspec
+
+    - name: Determine version
+      run: echo "VERSION=$(ls newrelic-telemetry_sdk-*.gem | ruby -pe 'sub(/newrelic-telemetry_sdk\-(.*).gem/, "\\1")')" >> $GITHUB_ENV
```
Comment (mwlang) [inline .github/workflows/release.yml:31]: Does this actually work?  Compared to the Ruby Agent's workflow: 
`echo ::set-env name=VERSION::$(ls newrelic_rpm-*.gem | ruby -pe 'sub(/newrelic_rpm\-(.*).gem/, "\\1")')`

We're calling `echo ::set-env name=VERSION` so doesnt't the above just echo the version to STDOUT?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      run: |
+        if [ $(git tag -l ${{ env.VERSION }}) ]; then
+          echo "Tag already created for this version"
+        else
+          git tag ${{ env.VERSION }}
+          git push origin ${{ env.VERSION }}
+        fi
+
+    - name: Obtain OTP to publish newrelic-telemetry_sdk to rubygems.org
+      run: echo "RUBYGEMS_OTP=$(python ./.github/workflows/scripts/rubygems-authenticate.py RUBYGEMS_MFA_KEY)"  >> $GITHUB_ENV
```
Comment (mwlang) [inline .github/workflows/release.yml:43]: Same concern as VERSION above.

File: .github/workflows/release.yml (hunk 1 above)
Comment (tannalynn) [reply .github/workflows/release.yml:31]: Actually, it's echoing into $GITHUB_ENV which is the new preferred way to set environment variables. I tested this part on my fork [here](https://github.com/tannalynn/newrelic-telemetry-sdk-ruby/runs/1322436764?check_suite_focus=true).

**Review by mwlang — DISMISSED** [review]
LGTM

**Comment by mwlang** [comment]
FYI, I cherry-picked the Snyk bundler fix from the documentation branch into this one.

**Review by rachelleahklein — APPROVED** [review]
Looks good! Thanks for setting this up, @tannalynn!

**Pull request merged**